src/packet/BinaryWriter.js

```diff
@@ -1,137 +1,62 @@
-﻿'use strict';
+'use strict';
 /*
  * Simple BinaryWriter is a minimal tool to write binary stream with unpredictable size.
  * Useful for binary serialization.
  *
- * Copyright (c) 2016 Barbosik
- * 
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- * 
- *     http://www.apache.org/licenses/LICENSE-2.0
- * 
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * 
- * 
+ * Copyright (c) 2016 Barbosik https://github.com/Barbosik
+ * License: Apache License, Version 2.0
  */
 
-function BinaryWriter(size) {
-    if (!size || size <= 0) {
-        size = Buffer.poolSize / 2;
-    }
-    this._buffer = new Buffer(size);
+function BinaryWriter() {
+    this._buffer = parseInt(process.version[1]) < 6 ? new Buffer(1e5) : Buffer.allocUnsafe(1e5);

```

Comment:
Anyway, I am done with this discussion. I have no respect for you kids except for @Megabyte918 , as he is showing great progress. You guys, didnt make much progress in life, you are still working on ogar after you people have worked on it for 5 years! I respect @Megabyte918 because he is showing progress. He is learning things he didn't know and is putting it to use. But you people, you claim you know everything. But my rant is: THEN SHOW IT!


I did not want to insult you peepz, but you have pushed my buttons.


@MastaCoder You claimed you have done much. WHERES THE PROOF!!

You have worked on:

1. Ogar
2. Cigar
3. MultiOgar
4. PolarsBots

THATS IT! And your dev group, doesnt do anything as well.  You were here for over 5 years! You existed before my account did! And yet you show no good progress.


@Luka967 You have some stuff on your profile. BUT IF I TAKE A CLOSER LOOK, THEY ARE NOTHING!

1. Take out all game-related stuff and you have a empty page

You, have also been around longer than I have. And yet you show no progress as well. 


You guys, are not learning anything new, but are instead wasting your time away with a game. STOP WASTING YOUR LIFE!

Meanwhile:

@Megabyte918 Started as a newbie, but became a good programmer in a year.

1. Manages MultiOgar
2. Always learning new things

To this, it is GOOD JOB

@Andrews54757

1. Was complete newbie but became advanced at js in half a year.
2. Started with OgarUL.
3. Now creates numerous modules and tools
4. Learning new stuff, such as cryptology.
5. Learned php and other things in a month.
6. Learning other languages for complex projects
7. Invented several algorithms. 
8. Learning web design methods.
9. Now starting a complex web site called codersAssemble
10. Other things such as calculator programs and 3d modeling.